include/imex/Dialect/GPUX/IR/GPUXOps.td

```diff
@@ -0,0 +1,183 @@
+        }];
+
+    // The C++ namespace that the dialect class definition resides in.
+    let cppNamespace = "::imex::gpux";
+}
+
+// Base class for dialect operations. This operation inherits from the base
+// `Op` class in OpBase.td, and provides:
+//   * The parent dialect of the operation.
+//   * The mnemonic for the operation, or the name without the dialect prefix.
+//   * A list of traits for the operation.
+class GPUX_Op<string mnemonic, list<Trait> traits = []> :
+    Op<GPUX_Dialect, mnemonic, traits>;
+
+// Provide a definition for the GPUX Stream/Device/Context Type for use in ODS. This allows for
+// using these types in a similar way to Tensor or MemRef. We use `DialectType`
+// to demarcate that these types belong to the GPUX dialect.
+def GPUX_StreamType
+    : DialectType<GPUX_Dialect, CPred<"$_self.isa<::imex::gpux::StreamType>()">,
+                  "Stream Type">;
+
+def GPUX_DeviceType
+    : DialectType<GPUX_Dialect, CPred<"$_self.isa<::imex::gpux::DeviceType>()">,
+                  "Device Type">;
+
+def GPUX_ContextType
+    : DialectType<GPUX_Dialect, CPred<"$_self.isa<::imex::gpux::ContextType>()">,
+                  "Context Type">;
+
+def GPUX_CreateDeviceOp : GPUX_Op<"create_device"> {
+
+  // Operation to create and return a device.
+  let arguments = (ins GPUX_DeviceType : $device);
+  let results = (outs GPUX_DeviceType : $gpu_device);
+}
+
+def GPUX_CreateContextOp : GPUX_Op<"create_context"> {
+
+  // Operation to create and return a context.
+  let arguments = (ins GPUX_DeviceType : $device);
+  let results = (outs GPUX_ContextType : $gpu_context);
+}
+
+def GPUX_CreateStreamOp : GPUX_Op<"create_stream", [AttrSizedOperandSegments]> {
+
+  // Operation to create and return a stream, based on the context
+  // and device. This stream is used for launching/queuing kernels
+  // on the GPU. If no device and context are provided, a default
+  // device and context will be created.
+  let arguments = (ins Optional<GPUX_DeviceType> : $device,
+                       Optional<GPUX_ContextType> : $context);
+  let results = (outs GPUX_StreamType : $gpux_stream);
+}
+
+def GPUX_DestroyDeviceOp : GPUX_Op<"destroy_device"> {
+
+  // Operation to deallocate the passed in device pointer.
+  let arguments = (ins GPUX_DeviceType : $device);
+}
+
+def GPUX_DestroyContextOp : GPUX_Op<"destroy_context"> {
+
+  // Operation to deallocate the passed in context pointer.
+  let arguments = (ins GPUX_ContextType : $context);
+}
+
+def GPUX_DestroyStreamOp : GPUX_Op<"destroy_stream"> {
+
+  // Operation to deallocate the passed in stream pointer.
+  let arguments = (ins GPUX_StreamType : $gpux_stream);
+}
+
+// The following ops are same as defined in upstream GPU dialect
+// https://github.com/llvm/llvm-project/blob/09c2b7c35af8c4bad39f03e9f60df8bd07323028/mlir/include/mlir/Dialect/GPU/GPUOps.td
+// with an additional argument for stream.
+
+def GPUX_LaunchFuncOp
```

Comment:
Should this be named LaunchFunc or LaunchKernel?

Reply:
followed what upstream Op had since its the same with an extra argument